sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/http/policy/BearerTokenAuthenticationChallengePolicy.java

```diff
@@ -32,70 +21,41 @@
 public class BearerTokenAuthenticationChallengePolicy implements HttpPipelinePolicy {
     private static final String AUTHORIZATION_HEADER = "Authorization";
     private static final String BEARER = "Bearer";
-    public static final Pattern AUTHENTICATION_CHALLENGE_PATTERN =
-        Pattern.compile("(\\w+) ((?:\\w+=\".*?\"(?:, )?)+)(?:, )?");
-    public static final Pattern AUTHENTICATION_CHALLENGE_PARAMS_PATTERN =
-        Pattern.compile("(?:(\\w+)=\"([^\"\"]*)\")+");
     public static final String WWW_AUTHENTICATE = "WWW-Authenticate";
-    public static final String CLAIMS_PARAMETER = "claims";
 
-    private final TokenCredential credential;
-    private final String[] scopes;
-    private final Supplier<Mono<AccessToken>> defaultTokenSupplier;
-    private final AccessTokenCache cache;
+    private final AccessTokenCacheImpl cache;
 
     /**
      * Creates BearerTokenAuthenticationChallengePolicy.
      *
      * @param credential the token credential to authenticate the request
-     * @param scopes the scopes of authentication the credential should get token for
      */
-    public BearerTokenAuthenticationChallengePolicy(TokenCredential credential, String... scopes) {
+    public BearerTokenAuthenticationChallengePolicy(TokenCredential credential) {
         Objects.requireNonNull(credential);
-        this.credential = credential;
-        this.scopes = scopes;
-        this.defaultTokenSupplier = () -> credential.getToken(new TokenRequestContext().addScopes(scopes));
-        this.cache = new AccessTokenCache(defaultTokenSupplier);
+        this.cache = new AccessTokenCacheImpl(credential);
     }
 
     /**
-     *
      * Executed before sending the initial request and authenticates the request.
      *
      * @param context The request context.
      * @return A {@link Mono} containing {@link Void}
      */
     public Mono<Void> onBeforeRequest(HttpPipelineCallContext context) {
```

Comment:
In .NET we landed on calling this `AuthenticateRequestAsync` and calling so it would be clear to subclasses which override it what their implementation should do. Because of this we also renamed the helper method `AuthenticateRequest` to `SetAuthorizationHeader`.